branch.go

```diff
@@ -28,27 +29,97 @@ func (m branchModule) Register(g *gin.RouterGroup) {
 
 // getProjectBranchListHandler godoc
 // @id getProjectBranchList
-// @summary Get list of branches. (NOT IMPLEMENTED!)
+// @summary Get list of branches.
 // @tags branch
 // @param projectId path uint true "project ID" minimum(0)
-// @success 501 "Not Implemented"
+// @success 200 {object} response.PaginatedBranches "Branches"
+// @failure 400 {object} problem.Response "Bad request"
+// @failure 401 {object} problem.Response "Unauthorized or missing jwt token"
+// @failure 404 {object} problem.Response "Project not found"
+// @failure 502 {object} problem.Response "Database is unreachable"
 // @router /project/{projectId}/branch [get]
 func (m branchModule) getProjectBranchListHandler(c *gin.Context) {
-	c.Status(http.StatusNotImplemented)
+	projectID, ok := ginutil.ParseParamUint(c, "projectId")
+	if !ok {
+		return
+	}
+	if !checkProjectExistsByID(c, m.Database, projectID, "when fetching list of branches for project") {
+		return
```

Comment:
I would prefer a recoverable crash/http500 to silently ignored errors. This seems to be missing log/error output.

I find some of the most difficult to debug errors are the ones where something is expected to happen but nothing actually happens followed by a code 0 style exit.

Reply:
There are other similar cases of this further down the file.

Reply:
I guess the naming is misleading. It does write a `problem.Response` to the HTTP response.

Maybe instead have "ensure" instead of "check"? Like `ensureProjectExistsByID`?

Reply:
Ahh i see what hapens here the funtion that gets called by the function that gets called in the if statement does logging. There is some abstract rule about never doing writes in the check for an if- i think it falls under idempotency. Im not sure logging qualifies here. Atleast in the case of a large application it probably would not. 

The name is probably fine. Although i guess your right- there is something to the word clarifying the boolean type of the return type that describes how much a function does. I guess i would expect a function named `validateProject(...)` (perhaps with a polymorphics signature accepting `int id`, or `Project proj`) to return state valid vs not valid and perhaps do a db read. While i would expect a function such as `isExsistProject(...)` || `isProject(...)`  to be more like a null or type check.

Reply:
I changed to `validate...` instead. Is that sufficient? Or is it still too unclear?

Reply:
I like it :)